pkg/operation/botanist/controlplane.go

```diff
@@ -1054,52 +1058,101 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
 		"storageCapacity":   b.Seed.GetValidVolumeSize("10Gi"),
 	}
 
-	etcd, err := b.InjectSeedShootImages(etcdConfig, common.ETCDImageName)
+	values, err := b.InjectSeedShootImages(defaultValues,
+		common.ETCDImageName,
+		common.ETCDBackupRestoreImageName,
+	)
 	if err != nil {
 		return err
 	}
 
+	values["prefix"] = common.GenerateBackupEntryName(b.Shoot.Info.Status.TechnicalID, b.Shoot.Info.Status.UID)
+
 	for _, role := range []string{common.EtcdRoleMain, common.EtcdRoleEvents} {
-		etcd["role"] = role
+		values["role"] = role
+		values["backupEnabled"] = false
+		values["provider"] = ""
 		if role == common.EtcdRoleMain {
 			// etcd-main emits extensive (histogram) metrics
-			etcd["metrics"] = "extensive"
+			values["metrics"] = "extensive"
+			if b.Seed.Info.Spec.Backup != nil {
+				values["provider"] = b.Seed.Info.Spec.Backup.Provider
+				values["backupEnabled"] = true
+				secret := &corev1.Secret{}
+				if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, common.BackupSecretName), secret); err != nil {
+					return err
+				}
+				values["container"] = string(secret.Data[common.BackupBucketName])
+			}
 		}
 
 		foundEtcd := true
-		statefulset := &appsv1.StatefulSet{}
-		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), statefulset); err != nil && !apierrors.IsNotFound(err) {
+		etcd := &druidv1alpha1.Etcd{}
+		if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), etcd); client.IgnoreNotFound(err) != nil {
 			return err
 		} else if apierrors.IsNotFound(err) {
 			foundEtcd = false
 		}
 
-		if foundEtcd && hvpaEnabled {
+		foundStatefulset := true
+		statefulset := &appsv1.StatefulSet{}
+		if foundEtcd {
+			if etcd.Status.Etcd.Name != "" {
+				if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, etcd.Status.Etcd.Name), statefulset); client.IgnoreNotFound(err) != nil {
+					return err
+				} else if apierrors.IsNotFound(err) {
+					foundStatefulset = false
+				}
+			}
+		} else {
+			// In case etcd resource is not deployed but statefulset exists. We w
+			if err := b.K8sSeedClient.Client().Get(ctx, kutil.Key(b.Shoot.SeedNamespace, fmt.Sprintf("etcd-%s", role)), statefulset); client.IgnoreNotFound(err) != nil {
+				return err
+			} else if apierrors.IsNotFound(err) {
+				foundStatefulset = false
+			}
+		}
+
+		if foundStatefulset && hvpaEnabled {
```

Comment:
When hvpa-controller gets CRD support, both gardener and hvpa-controller would be reconciling `etcd` resource using the statefulset which could result in a race.
Can we please put a `TODO` here because this will need to be revisited when hvpa gets CRD support.

Reply:
Done